utils/mentions.py

```diff
@@ -0,0 +1,46 @@
+import re
+from enum import Enum
+
+from discord.ext.commands import Converter
+
+from models import db_session
+from models.user import User
+
+__all__ = ["MentionConverter"]
+
+
+class MentionType(Enum):
+    ID = (0,)
+    STRING = 1
+
+
+class Mention:
+    def __init__(self, type, id, string):
+        self.type: MentionType = type
+        self.id: int = id
+        self.string: str = string
+
+    def is_id_type(self):
+        return self.type == MentionType.ID
+
+    def type_str(self):
+        if self.type == MentionType.ID:
+            return "id"
+
+        return "string"
+
+
+def parse_mention(string, db_session=db_session) -> Mention:
+    if re.match("^<@!?\d+>$", string):
```

Comment:
Using regex to extract the mention is a bit nasty. You can make use of the existing discord [member converter](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.MemberConverter) to handle the legwork for you.

Reply:
I used the MemberConverter method in an earlier version, but figured I could use this method to make a non-async function (easier to integrate tests without figuring out how to implement async tests). As it turned out, [MemberConverter also uses a near-identical regex  implementation to identify the UID](https://github.com/Rapptz/discord.py/blob/c62822440368afc7124e35f761a78b6bf818c05e/discord/ext/commands/converter.py#L279).
Since the UID is the only thing we're interested in here, I'm not sure using MemberConverter is particularly more useful.